tests/end-to-end/upgrade/pkg/tests/function/function.go

```diff
@@ -0,0 +1,188 @@
+package function
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+	"strings"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubelessV1 "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	kubeless "github.com/kubeless/kubeless/pkg/client/clientset/versioned"
+	"github.com/sirupsen/logrus"
+)
+
+type FunctionUpgradeTest struct {
+	functionName, uuid string
+	kubelessClient     *kubeless.Clientset
+	coreClient         *kubernetes.Clientset
+}
+
+func NewFunctionUpgradeTest() (*FunctionUpgradeTest) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return &FunctionUpgradeTest{}
+	}
+
+	kubelessClient, err := kubeless.NewForConfig(config)
+	if err != nil {
+		return &FunctionUpgradeTest{}
+	}
+
+	coreClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return &FunctionUpgradeTest{}
+	}
+
+	return &FunctionUpgradeTest{
+		kubelessClient: kubelessClient,
+		coreClient:     coreClient,
+		functionName:   "hello",
+		uuid:           uuid.New().String(),
+	}
+}
+
+func (f *FunctionUpgradeTest) CreateResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	log.Println("FunctionUpgradeTest creating resources")
+	_, err := f.createFunction(log, namespace)
+	if err != nil {
+		return err
+	}
+
+	<-stop
```

Comment:
the stop channel is used to stop the long process, for example creating function process is being executed for 10 minutes and the framework decides to stop it, so do not read from the channel (the same for TestResources